### ISSUE TITLE
sed: read error on /etc/upstream-release: Is a directory

### ISSUE THREAD

**Issue by ruthan** [benjamimgois/goverlay#139]
Hi,
i compiled and installed last mangohud on Mint22, but its not working and goverlay (from Software center) is not starting with this strange error:
sed: read error on /etc/upstream-release: Is a directory
  I reinstalled it too.

**Comment by benjamimgois** [comment]
Hi @ruthan , please do not use the goverlay version from the repositories. Use the latest github release.

**Comment by ruthan** [comment]
Well if repository version is not good, you should add some warning or somelike like that if you cant stop to add old ones to repositories Simple version compatibility check a check how old package is. Github version also most means not great user experience, lots of programs not have easy to install packages. 
  I have to yet check how about Goverlay. I already compiled mangohud from git, but it seems that gooverlay from repository overwrited it without some prompt. It actually was worse, because old Mangohud worked with Goverlay and new doesnt. I really dont like Linux solution of these things in case of Windows you just download installer, click next, next and its done, or app can autoupdate itself. 
   Well there is snap, but its sort of bloatware.. Linux, Linux has based on sharing libs, which should save a space, snap or flatpack is fixing it, but these packages are huge, much bigger that Windows programs which are not care about sharing and space at all , there is something rotten in that..I usually get repository app version like 20 MB and flatpack like 1GB for simple utility.

**Comment by benjamimgois** [comment]
@ruthan i agree that universal software distribution isn't ideal yet, this has being a old issue on Linux, but i belive flatpak is working towards this. Ubuntu and Mint are distros well known for it ease of use for home and office, but i do not think that they are the best fit for gaming. I would recommend you to give a try a distro named "Manjaro", it has a much more update kernel and drivers, also the repositories are also very updated. You can install anything with a few clicks.

https://manjaro.org/

**Comment by ruthan** [comment]
I think that problem is developers side too. If you compare average user app experience on Win+Mac, Linux is still very different. Simply state of program which is considered good enough for user is quite a different. That is way people preferring multiplatform app ported to Linux from other platforms, is not secret or some magic.
 
    There are quite a few apps which made it from Linux to Windows the same way. 
    As i wrote flatpack only and main problem is that size is often 10x, 50x bigger that version from repository, otherwise would use only them.

   I have 12 or 14 Linux distros on my machines, im Using Linux/Unix 20+ years, SteamOS is nice for gamming, but desktop experience is not great. MX Linux seems to be new rising star. I have Manjaro too, but its Arch based (i have Arch too), which is probably the most complex distro, so it has disadvatage too (for example Nvidia driver autogenerate Xinerama line, which is working with all other distros could kill whole X ). I dont mind that some packages are a bit older, its good price for overall better user experience. New releases should be more about adding new  features than mainly fixing not great previous releases. 

  I already wrote some easy ways how to make it better with quite small time effort, i allways tried to make my programs much fail proof as possible, even if it was for very few users, it always saved a lot of time:
  - add new version check, compatibility check and notification and release notes as check as part of program package is easy.. more advanced step is make downloading and upgrading of program as part of app itself.
  - add some error messaging and handling.. Its better write, sorry package is too old not compatible.. is much better than  _/etc/upstream-release: Is a directory.
   - make major distros packages, to be honest they are only 3 - it means Debian / Arch / Fedora. 
   - Mangohud is good example, there is not even some -H / help output, to even link to github description.. and github description lot of details, destription.. there are not even error message when Mangohud is not working with some app.. there is not compatibility info- without could be build with community support, if you make some basic framework.

  I really think that problem is overall mindset, that user should study app or OS internal workings to use app. I see OS as platform to run apps and it should require minimum effort, unless you need something not default, advanced and if all is intuitive and fool prove, it same a lot of time and user can use time to things which really have to be complex.
